_includes/layouts/base.html

```diff
@@ -12,7 +12,14 @@
 <body>
   {% include "header.html" %}
 
-  {% include "menu.html" primary_navigation: site.primary_navigation secondary_navigation: site.secondary_navigation %}
+
+
+  {% case navigation %}
+  {% when "pages" %}
+    {% include "menu.html" primary_navigation: pages.navigation.primary secondary_navigation: pages.navigation.secondary %}
+  {% else %}
+      {% include "menu.html" primary_navigation: site.primary_navigation  secondary_navigation: site.secondary_navigation %}
+  {% endcase %}
```

Comment:
we use assign in so many places, it seems obv to do it here too. would something like this work? note that you'd need to rename the final keys for one of them (`primary` -> `primary_navigation` or vice-vera )
```
{% assign nav = site %}
{% if "pages" %} {% nav = pages.navigation %} {% endif %}
{% include "menu.html" primary_navigation: nav.primary secondary_navigation: nav.secondary %}
  
```